tests/integration_tests/functional/test_metrics.py

```diff
@@ -23,4 +26,41 @@ def test_flush_metrics(test_microvm_with_api):
 
     microvm.start()
 
+    cmd = 'read line < {} && echo $line'.format(metrics_fifo_path)
+    res = utils.run_cmd(cmd)
+    metrics = json.loads(res.stdout)
+
+    assert metrics['put_api_requests'] == {
+      "actions_count": 1,
+      "actions_fails": 0,
+      "boot_source_count": 1,
+      "boot_source_fails": 0,
+      "drive_count": 1,
+      "drive_fails": 0,
+      "logger_count": 0,
+      "logger_fails": 0,
+      "machine_cfg_count": 1,
+      "machine_cfg_fails": 0,
+      "metrics_count": 1,
+      "metrics_fails": 0,
+      "network_count": 0,
+      "network_fails": 0
+    } and metrics['block'] == {
+      "activate_fails": 0,
+      "cfg_fails": 0,
+      "no_avail_buffer": 0,
+      "event_fails": 0,
+      "execute_fails": 0,
+      "invalid_reqs_count": 0,
+      "flush_count": 0,
+      "queue_event_count": 0,
+      "rate_limiter_event_count": 0,
+      "update_count": 0,
+      "update_fails": 0,
+      "read_bytes": 0,
+      "write_bytes": 0,
+      "read_count": 0,
+      "write_count": 0
+    }
```

Comment:
Can we check just a couple of individual metrics instead of groups ? For example:

`metrics['put_api_requests']['actions_count'] == 1`
`metrics['put_api_requests']['metrics_count'] == 1`

Reply:
I think the grouping approach is more useful since it will force code that adds new metrics to also update this test. My proposal would be to create a separate json in our CI folder that holds the expected json output (just like the one located in tests/framework/config_vm.json), load it in the test and check it against the one obtain  by the test. WDYT?

Reply:
I agree with @dianpopa 's comment, which indicates my idea for this pr. I actually want to include more groups to deeply check if the metrics values is as excepted under integration test cases. But so far I only put 2 important and expectable group to this pr. Maybe we can cover almost all groups  in future.

>> My proposal would be to create a separate json in our CI folder that holds the expected json output (just like the one located in tests/framework/config_vm.json), load it in the test and check it against the one obtain by the test. WDYT?

I had considered this, but 1. some groups value may be un-exceptable, so the json will seem to be incomplete;  2. we maybe add more process to metrics test, and have to compare with different groups or values, so currently pasting the expected values to code maybe looks  more clear. WDYT?

Reply:
My concern is that this test would end up being very complex if we want to check all the metrics values here. Also it would be responsible to check some things that are not really related to the metrics system. For example I think that checking that the block metrics values are correct is the responsibility of the block related tests, rather then the responsibility of this test. An option would be to disregard the values here, and just check that all the metrics are printed. And maybe we could also check that not all are equal to 0. What do you think ?

Reply:
@serban300 , this sounds great! We can just check the metrics output keys, with disregarding their values.

Reply:
@serban300 @dianpopa Updated, PTAL.

Reply:
I change to check only metrics root keys are as expected. The nested keys (like inside `block`, `put_api_requests `...) should be put on the relative module test.

Reply:
Looks good to me. Thanks for doing the changes !